darts/utils/likelihood_models.py

```diff
@@ -35,64 +153,852 @@ def _sample(self, model_output: torch.Tensor) -> torch.Tensor:
 
     @property
     @abstractmethod
-    def _num_parameters(self) -> int:
+    def num_parameters(self) -> int:
         """
         Returns the number of parameters that define the probability distribution for one single
         target value.
         """
         pass
 
 
-class GaussianLikelihoodModel(LikelihoodModel):
-    """
-    Gaussian Likelihood
-    """
-    def __init__(self):
-        self.loss = nn.GaussianNLLLoss(reduction='mean')
+class GaussianLikelihood(Likelihood):
+    def __init__(self, prior_mu=None, prior_sigma=None, prior_strength=1.):
+        """
+        Univariate Gaussian distribution.
+
+        https://en.wikipedia.org/wiki/Normal_distribution
+
+        - Univariate continuous distribution.
+        - Support: :math:`\mathbb{R}`.
+        - Parameters: mean :math:`\\mu \in \mathbb{R}`, standard deviation :math:`\\sigma > 0`.
+
+        Parameters
+        ----------
+        prior_mu
+            mean of the prior Gaussian distribution (default: None).
+        prior_sigma
+            standard deviation (or scale) of the prior Gaussian distribution (default: None)
+        prior_strength
+            strength of the loss regularisation induced by the prior
+        """
+        self.prior_mu = prior_mu
+        self.prior_sigma = prior_sigma
+        _check_strict_positive(self.prior_sigma, 'sigma')
+
+        self.nllloss = nn.GaussianNLLLoss(reduction='mean', full=True)
         self.softplus = nn.Softplus()
-        super().__init__()
 
-    def _compute_loss(self, model_output: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
-        model_output_means, model_output_vars = self._means_and_vars_from_model_output(model_output)
-        return self.loss(model_output_means.contiguous(), target.contiguous(), model_output_vars.contiguous())
+        super().__init__(prior_strength)
 
-    def _sample(self, model_output: torch.Tensor) -> torch.Tensor:
-        model_output_means, model_output_vars = self._means_and_vars_from_model_output(model_output)
-        return torch.normal(model_output_means, model_output_vars)
+    def _nllloss(self, params_out, target):
+        means_out, sigmas_out = params_out
+        return self.nllloss(means_out.contiguous(), target.contiguous(), sigmas_out.contiguous())
```

Comment:
I remember I had to add the `.contigous` for it to work, but is this still the case? because it is only used here.

Reply:
I actually tried to remove it, but it crashes. I didn't investigate it, but it's a bit weird (almost looks like it could be a bug in PyTorch?). In any case let's leave the contiguous() for now.